File: Zarina/TaZQL/src/ast/ID.java
package ast;

public class ID extends Testik {

	private String id;
	
	public ID(String id) {
		this.id = id;
	}
	
	public String getID() {
		return id;
	}
/*
	@Override
	public String toString() {
		return id;
	}
*/	
	@Override
	public <T> T accept(IMainVisitor<T> visitor) {
		return visitor.visit(this);
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Computation/Id.cs
using AST.Nodes.Interfaces;
using AST.Representation;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AST.Nodes.Computation
{
    public class Id : ASTNode, IComputation
    {
        public string Value {get; private set;}
        private string parsedString;

        public Id(string parsedString, string value , PositionInText positionInText) 
            : base(positionInText)
        {
            this.Value = value;
            this.parsedString = parsedString;
        }
        public override string GetParsedString()
        {
            return parsedString;
        }

        public override void Accept(Visitors.IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public override T Accept<T>(Visitors.IVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }
    }
}
